dds/DCPS/DataReaderImpl.cpp

```diff
@@ -308,7 +308,9 @@ DataReaderImpl::add_association(const RepoId& yourId,
 
   //get message block from octet seq then deser to a type info
   XTypes::TypeInformation ti;
-  XTypes::deserialize_type_info(ti, writer.serializedTypeInfo);
+  DCPS::OctetSeq seq;
+  copy(seq, writer.serializedTypeInfo);
```

Comment:
Looks costly to each time copy the buffer just to pass it to a different operation which expects a OctetSeq in a different namespace

Reply:
It's not ideal but this is not "critical path"